apps/namadillo/src/App/NamadaTransfer/NamadaTransfer.tsx

```diff
@@ -196,21 +123,10 @@ export const NamadaTransfer: React.FC = () => {
     try {
       setGeneralErrorMessage("");
 
-      if (typeof sourceAddress === "undefined") {
-        throw new Error("Source address is not defined");
-      }
-
-      if (!chainId) {
-        throw new Error("Chain ID is undefined");
-      }
-
-      if (!selectedAsset) {
-        throw new Error("No asset is selected");
-      }
-
-      if (typeof gasConfig === "undefined") {
-        throw new Error("No gas config");
-      }
+      invariant(sourceAddress, "Source address is not defined");
+      invariant(chainId, "Chain ID is undefined");
```

Comment:
Maybe, just maybe, we could use a function to validate these variables here? We're repeating this validation multiple times

Reply:
Yeah, I agree, some variables on the begin of the component as well. I'll explore this more on the next PRs